content/learn/keyconcepts.adoc

```diff
@@ -0,0 +1,164 @@
+---
+layout: default
+title: Key concepts
+menu: learn
+weight: 20
+aliases: /key concepts/
+---
+
+:toc:
+
+:imagesdir: /images
+:_content-type: ASSEMBLY
+include::modules/comm-attributes.adoc[]
+
+= Key concepts
+
+This section describes a set of key concepts essential for creating validated pattern. This material provides you with foundational knowledge to get familiar with the validated patterns framework. The content offers a clear and structured starting point, helping you understand and effectively apply these concepts in your projects.
+
+[id="helm-and-kustomize"]
+== Helm and Kustomize
+
+Two options are available that allow you to manage the lifecycle of Kubernetes applications you want to deploy with the validated patterns framework. The tools supported are `Kustomize` and `Helm`. 
+
+* *Kustomize*: Kustomize is a configuration management tool used to make declarative changes to application configurations and components and preserve the original base YAML files. Kustomize allows you to manage Kubernetes resource files in a declarative manner, using a set of layered configurations. This makes it easier to maintain, version, and apply changes to Kubernetes resources without having to modify the original YAML files directly.
+* *Helm*: Helm is a package manager for Kubernetes, which simplifies the process of defining, installing, and upgrading complex Kubernetes applications. It manages Kubernetes manifests using a packaging format called `Charts`.
```

Comment:
All true, yet somehow I feel this misses the mark.

All the things written about kustomize are also true of helm.
Which leads to the general observation that they are both fundamentally templating tools, just with a different emphasis.

kustomize excels at problems like "I found a set of yamls I want, but put them in namespace X, prefix their names with Y, and set all the image tags to Z" but X, Y, and Z must all be pre-defined and stored in git as "overlays".  This leads to a proliferation of overlay directories (if N devs need an overlay for M components, that scales poorly) and they quickly become both disorganise and the increasing layers of indirection make them non-trivial to read.

Instead of ad-hoc patching of yaml files, helm provides a templating language.  X, Y, and Z remain as variables in the raw yaml, and helm combines the templates with the actual values at runtime.  Additionally, the language provides the power of conditionals, so the chart can adapt to its environment (eg. AWS vs GCP), and loops - which no longer means every instance of a given type needs to be defined explicitly, and that they can all be created consistently.  You can for example ask for a list of 12 namespaces.